.github/workflows/e2e-branch.yml

```diff
@@ -0,0 +1,169 @@
+    strategy:
+      matrix:
+        project: ${{ fromJSON(needs.install-deps.outputs.projects) }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '14'
+      - name: Retrieve dependencies cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            node_modules
+            !node_modules/.cache/nx
+            /home/runner/.cache/Cypress
+          key: cache-pr-preview-${{ hashFiles('package-lock.json') }}
+      - name: Retrieve nx cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            .angular
+            node_modules/.cache/nx
+            dist
+          key: cache-nx-storybook-${{ matrix.project }}-${{ hashFiles('package-lock.json') }}
+          restore-keys: |
+            cache-nx-storybook-${{ matrix.project }}-
+            cache-nx-storybook-
+            cache-nx-
+      - name: Build ${{ matrix.project }}
+        run: |
+          npx nx run ${{ matrix.project }}:build-storybook:ci --outputDir=dist/storybooks/${{ matrix.project }}
+          npx sb extract ./dist/storybooks/${{ matrix.project }}
+      - name: Upload storybook artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: ${{ matrix.project }}
+          path: ./dist/storybooks/${{ matrix.project }}
+          retention-days: 1
+          if-no-files-found: error
+
+  publish:
+    name: Publish
+    runs-on: ubuntu-latest
+    needs:
+      - install-deps
+      - build-storybook
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '14'
+      - name: Retrieve dependencies cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            node_modules
+            !node_modules/.cache/nx
+            /home/runner/.cache/Cypress
+          key: cache-pr-preview-${{ hashFiles('package-lock.json') }}
+      - name: Retrieve nx cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            .angular
+            node_modules/.cache/nx
+            dist/storybook
+          key: cache-nx-publish-${{ hashFiles('package-lock.json') }}
+          restore-keys: |
+            cache-nx-publish-
+            cache-nx-
+      - name: Download artifacts
+        uses: actions/download-artifact@v3
+        with:
+          path: ./dist/storybooks/{{ github.ref_name }}/
+      - name: Generate Storybook Composition
+        run: npx nx g @skyux/e2e-schematics:storybook-composition --projectsJson '${{ needs.install-deps.outputs.projects }}' --baseUrl ../storybooks/${{ github.ref_name }}/
+      - name: Build Storybook Composition
+        run: npx nx run storybook:build-storybook:ci --outputDir=dist/storybook
+      - name: Checkout ${{ env.GH_PAGES_REPO }}
+        uses: actions/checkout@v3
+        with:
+          repository: ${{ env.GH_PAGES_OWNER }}/${{ env.GH_PAGES_REPO }}
+          ref: 'main'
```

Comment:
Is `ref: 'main'` needed? Would we need to change this value once we start running this workflow against the `6.x.x` branch?